pkg/dns/vips_allocator.go

```diff
@@ -128,6 +130,39 @@ func (d *VIPsAllocator) createOrUpdateMeshVIPConfig(
 	return d.persistence.Set(ctx, mesh, out)
 }
 
+func addFromMeshGateway(outboundSet *vips.VirtualOutboundMeshView, mesh string, gateway *core_mesh.MeshGatewayResource) {
+	for i, listener := range gateway.Spec.Conf.Listeners {
+		// We only setup outbounds for cross mesh listeners and only ones with a
+		// concrete hostname.
+		if !listener.CrossMesh || strings.Contains(listener.Hostname, "*") {
+			continue
+		}
+
+		for _, selector := range gateway.Spec.Selectors {
+			tags := mesh_proto.Merge(
+				gateway.Spec.GetTags(),
+				listener.GetTags(),
+				map[string]string{
+					"kuma.io/mesh": mesh,
+				},
+				selector.GetMatch(),
+			)
+			// TODO validate only one selector
+			hostname := listener.Hostname
+			origin := fmt.Sprintf("mesh-gateway:%s:%s:%s", mesh, gateway.GetMeta().GetName(), hostname)
+
+			if err := outboundSet.Add(vips.NewHostEntry(hostname), vips.OutboundEntry{
```

Comment:
Shouldn't they result in the same host entry/outbound entry for all intents and purposes? I think their behavior is the same from the point of view of virtual outbounds.

Reply:
> I'm going to change it to Fqdn because it looks like Host is deprecated but I kind of like that it's the same type as a virtual outbound entry

Do we have an issue to remove `Host`? I think it's been deprecated for long enough no?
Using FQDN sounds like the right option yes

Reply:
Using FQDN now, it's kind of a shame though because `Host` added a mapping for port 80 which isn't otherwise possible without #4084